fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNavigationMiddleware.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState
+
+/**
+ * Middleware that handles navigation events for the review quality check feature.
+ *
+ * @property openLinkInNewTab Callback used to open an url in new tab.
+ * @property openLinkInSelectedTab Callback used to open a url in the selected tab.
+ * @property scope [CoroutineScope] used to launch coroutines.
+ */
+class ReviewQualityCheckNavigationMiddleware(
+    private val openLinkInNewTab: (String) -> Unit,
+    private val openLinkInSelectedTab: (String) -> Unit,
+    private val scope: CoroutineScope,
+) : Middleware<ReviewQualityCheckState, ReviewQualityCheckAction> {
+
+    override fun invoke(
+        context: MiddlewareContext<ReviewQualityCheckState, ReviewQualityCheckAction>,
+        next: (ReviewQualityCheckAction) -> Unit,
+        action: ReviewQualityCheckAction,
+    ) {
+        next(action)
+        when (action) {
+            is ReviewQualityCheckAction.OpenLink -> processAction(action)
+            else -> {
+                // no-op
+            }
+        }
+    }
+
+    private fun processAction(
+        action: ReviewQualityCheckAction.OpenLink,
+    ) = scope.launch {
+        when (action.link) {
```

Comment:
if we want to mirror the pattern in the other [middlewares](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt), we instead want to base the `invoke` `when` off of whether the action is `NavigationMiddlewareAction` and then use that type inside for `action` inside of `processAction`. 

```suggestion
        when (action) {
            is ReviewQualityCheckAction.NavigationMiddlewareAction -> processAction(action)
            else -> {
                // no-op
            }
        }
    }

    private fun processAction(
        action: ReviewQualityCheckAction.NavigationMiddlewareAction,
    ) = scope.launch {
        when (action) {
```